### ISSUE TITLE
y-axis DOS ticks

### ISSUE THREAD

**Issue by ajjackson** [SMTG-UCL/sumo#35]
> You did  a nice job. I am having an issue with DOS plot, the issue is including y-axis tick in the plot.
> thank you.

_Originally posted by @mukhtarbayerouniversity in https://github.com/SMTG-UCL/sumo/issues/18#issuecomment-432668609_

**Comment by ajjackson** [comment]
This is something we turned off because we don't find it to be useful or pretty. But it should be possible to work around these differences in taste by using Matplotlib style sheets!

You need to change line 291 of *sumo/plotting/dos_plotter.py* to
```ax.tick_params(axis='y', labelleft='off')```
(with the same indentation as before). This will stop the ticks being removed within the plotter itself, so it becomes strictly a style issue.

You can now create a style file called e.g. "dos_ticks.mplstyle" containing
```
ytick.right : True
ytick.left : True
```
and call dosplot with
```sumo-dosplot --style dos_ticks.mplstyle```.

If @utf doesn't see any problems with this then we can add that change to the next release.

**Comment by ajjackson** [comment]
I'm labelling this "bug" because this fix should work without having to edit any code.

**Comment by utf** [comment]
I think it's reasonable that users should be able to turn on the ticks if they wish. @ajjackson I'll make the change that you suggested.

**Comment by mukhtarbayerouniversity** [comment]
thanks for the reply. But i tried your suggesstion, it's still not working.

**Comment by ajjackson** [comment]
Did you change *sumo/plotting/dos_plotter.py* in the place it is installed? Try
```
 python3 -c "import sumo.plotting.dos_plotter; print(sumo.plotting.dos_plotter.__file__)"
```
to find the location of this file.

**Comment by mukhtarbayerouniversity** [comment]
Thanks @ajjackson, i did all the necessary steps all this while. It did not work. below is the responds i got when i tired your las suggestion (i have already edited the bandplot script as suggested earlier)

 
python3 -c "import sumo.plotting.dos_plotter; print(sumo.plotting.dos_plotter._dos_ticks.mplstyle_)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: module 'sumo.plotting.dos_plotter' has no attribute '_dos_ticks'

**Comment by ajjackson** [comment]
```python3 -c "import sumo.plotting.dos_plotter; print(sumo.plotting.dos_plotter.dos_ticks.mplstyle)"```
That's not what I wrote; the `__file__` is actually literal. It's a Python "dunder" method which can be used to find out where a module is installed.

*dos_ticks.mplstyle* would not be part of the Sumo package, it would be a user-generated style file so it should stay in user space. For example, the same directory as you are plotting in, or some general purpose documents folder. It is up to you to put the style sheet somewhere convenient for yourself and then point to it with
```sumo-dosplot --style /path/to/my/very/own/file.mplstyle```

**Comment by ajjackson** [comment]
You could remove that whole `ax.tick_params...` line in that case. I'm curious as to why that would be useful? These values are rarely labelled in academic papers.

**Comment by mukhtarbayerouniversity** [comment]
thanks alot for the help, yeah was just curious if its possible to add them.

**Comment by utf** [comment]
Fixed in f0b2ed6918480f83fb66c797e98ed7297ad74c8a
